pkg/controller/user-cluster-controller-manager/resources/reconciler.go

```diff
@@ -655,3 +663,13 @@ type reconcileData struct {
 	userSSHKeys   map[string][]byte
 	cloudConfig   []byte
 }
+
+func (r *reconciler) ensureOPAIntegrationIsRemoved(ctx context.Context) error {
+	for _, resource := range gatekeeper.GetResourcesToRemoveOnDelete() {
+		if err := r.Client.Delete(ctx, resource); err != nil && !errors.IsNotFound(err) {
+			return fmt.Errorf("failed to ensure OPA integration is removed/not present: %v", err)
```

Comment:
Would be cool to add also the name of the resource that's being deleted in the message